src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEngine.Unix.cs

```diff
@@ -185,21 +167,7 @@ private static void AllocateToken(SocketAsyncContext context, out SocketAsyncEng
                 engine = s_currentEngines[s_allocateFromEngine];
                 if (engine == null)
                 {
-                    // We minimize the number of engines on applications that have a low number of concurrent sockets.
-                    for (int i = 0; i < s_allocateFromEngine; i++)
-                    {
-                        var previousEngine = s_currentEngines[i];
-                        if (previousEngine == null || previousEngine.HasLowNumberOfSockets)
-                        {
-                            s_allocateFromEngine = i;
-                            engine = previousEngine;
-                            break;
-                        }
-                    }
-                    if (engine == null)
-                    {
-                        s_currentEngines[s_allocateFromEngine] = engine = new SocketAsyncEngine();
-                    }
+                    s_currentEngines[s_allocateFromEngine] = engine = new SocketAsyncEngine();
```

Comment:
FWIW, the above can be condensed to:
```C#
engine = s_currentEngines[s_allocateFromEngine] ??= new SocketAsyncEngine();
```

Reply:
Nit: pre-existing your code, but the `current` in `s_currentEngines` seems superfluous.  Maybe just `s_engines`.